sc/src/main/java/uk/ac/ebi/atlas/experimentpage/JsonExperimentTSnePlotController.java

```diff
@@ -125,6 +140,21 @@ public String tSnePlotWithExpression(
         return GSON.toJson(model);
     }
 
+    @RequestMapping(
+            value = "json/experiments/{experimentAccession}/metadata",
+            method = RequestMethod.GET,
+            produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    @ResponseBody
+    public String getTSnePlotMetadata(
+            @PathVariable String experimentAccession) {
+        JsonArray plotData = experimentPageContentService.getTsnePlotMetaData(experimentAccession);
+        if(plotData.size() == 0) {
+            return null;
```

Comment:
Adding this has caused your test to fail. Can you think why?

Reply:
Also, what is your reasoning for not letting Gson handle empty results?

Reply:
Can you please let me know which test case is failing?

In the case of empty result `GSON.toJson(plotData)` returns `[]` as a String which I don't require and to handle it in my empty metadata check I have to do a direct string manupilation. To avoid that I return null in case of an empty array.

Reply:
Have a look at [Bamboo](http://bar:8085/browse/ATLASOLD-SCXACI20-GITCO-1/test/case/57658993) for the failing test.